src/pages/MoviesList/MovieApp.jsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable prefer-const */
+import React, { useEffect, useState } from "react";
+import MovieList from "components/MovieList/MovieList";
+import Footer from "components/Footer/Footer";
+import SidePanel from "components/SidePanel/SidePanel";
+import { Container } from "pages/MoviesList/styles";
+import axios from "axios";
+import Header from "components/Header/Header";
+/**
+ * MovieApp Component
+ * Main component that represents the movie application.
+ */
+export default function MovieApp() {
+  const [movies, setMovies] = useState([]);
+  const [selectedGenres, setSelectedGenres] = useState([]);
+  const [availabilities, setAvailabilities] = useState([]);
+  const [selectedKeyword, setSelectedKeyword] = useState([]);
+
+  const [release, setRelease] = useState([]);
+  const [params, setParams] = useState({
+    page: 1,
+    region: "",
+    "release_date.gte": "",
+    "release_date.lte": "2024-03-21",
+    show_me: 0,
+    sort_by: "popularity.desc",
+    with_genres: "",
+    with_keywords: "",
+    with_original_language: "",
+    with_watch_monetization_types: "",
+    with_release_type: "",
+  });
+  /**
+   * Function to handle movie search and update parameters for API request.
+   * Fetches movies based on the updated parameters.
+   *
+   * @param {Event} e - The event object from the form submission.
+   */
+  const getMovies = (e) => {
+    let updatedParams = params;
```

Comment:
`    let updatedParams = {...params}; `
States in React should be immutable in order to get the expected behavior successfully. I encourage you to read the following article on why React is immutable: 
https://reacttraining.com/blog/state-in-react-is-immutable